File: MyPersonalIndex/Classes/Queries/PortfolioQueries.cs
using System;

namespace MyPersonalIndex
{
    class PortfolioQueries : Queries
    {
        public static string UpdatePortfolio(int Portfolio, string Name, bool Dividends, double NAVStart, int CostCalc, int AAThreshold, DateTime StartDate)
        {
            return string.Format("UPDATE Portfolios SET Name = '{0}', Dividends = {1}, NAVStartValue = {2}, CostCalc = {3}, AAThreshold = {4}, StartDate = '{5}' WHERE ID = {6}",
                Functions.SQLCleanString(Name), Convert.ToByte(Dividends), NAVStart, CostCalc, AAThreshold, StartDate, Portfolio);
        }

        public static string InsertPortfolio(string Name, bool Dividends, double NAVStart, int CostCalc, int AAThreshold, DateTime StartDate)
        {
            return string.Format("INSERT INTO Portfolios (Name, Dividends, NAVStartValue, CostCalc, AAThreshold, StartDate, HoldingsShowHidden, NAVSort, HoldingsSort, AASort, AAShowBlank, CorrelationShowHidden, AcctSort, AcctShowBlank)" +
                " VALUES ('{0}', {1}, {2}, {3}, {4}, '{5}', 1, 1, '', '', 1, 1, '', 1)",
                Functions.SQLCleanString(Name), Convert.ToByte(Dividends), NAVStart, CostCalc, AAThreshold, StartDate.ToShortDateString());
        }
    }
}

File: MyPersonalIndex/classes/queries/portfolioQueries.cpp
#include "portfolioQueries.h"
#include "queries.h"
#include "globals.h"

queries::queryInfo* portfolioQueries::updatePortfolio(globals::portfolio* const p)
{
    if (!p)
        return 0;

    if(p->id == -1) // insert new
    {
        return new queryInfo(
            "INSERT INTO Portfolios (Description, StartValue, AAThreshold, ThresholdMethod, CostCalc, StartDate, Dividends, HoldingsShowHidden,"
                " HoldingsSort, AAShowBlank, AASort, CorrelationShowHidden, AcctShowBlank, AcctSort, NAVSortDesc)"
            " VALUES (:Description, :StartValue, :AAThreshold, :ThresholdMethod, :CostCalc, :StartDate, :Dividends, :HoldingsShowHidden,"
                " :HoldingsSort, :AAShowBlank, :AASort, :CorrelationShowHidden, :AcctShowBlank, :AcctSort, :NAVSortDesc)",
            QList<parameter>()
                << parameter(":Description", p->description)
                << parameter(":StartValue", p->startValue)
                << parameter(":AAThreshold", p->aaThreshold)
                << parameter(":ThresholdMethod", (int)p->aaThresholdMethod)
                << parameter(":CostCalc", (int)p->costCalc)
                << parameter(":StartDate", p->origStartDate.toJulianDay())
                << parameter(":Dividends", (int)p->dividends)
                << parameter(":HoldingsShowHidden", (int)p->holdingsShowHidden)
                << parameter(":HoldingsSort", p->holdingsSort)
                << parameter(":AAShowBlank", (int)p->aaShowBlank)
                << parameter(":AASort", p->aaSort)
                << parameter(":CorrelationShowHidden", (int)p->correlationShowHidden)
                << parameter(":AcctShowBlank", (int)p->acctShowBlank)
                << parameter(":AcctSort", p->acctSort)
                << parameter(":NAVSortDesc", (int)p->navSortDesc)
        );
    }
    else // update
    {
        return new queryInfo(
            "UPDATE Portfolios SET Description = :Description, Dividends = :Dividends, StartValue = :StartValue, CostCalc = :CostCalc, AAThreshold = :AAThreshold,"
            " ThresholdMethod = :ThresholdMethod, StartDate = :StartDate WHERE ID = :PortfolioID",
            QList<parameter>()
                << parameter(":Description", p->description)
                << parameter(":Dividends", (int)p->dividends)
                << parameter(":StartValue", p->startValue)
                << parameter(":CostCalc", (int)p->costCalc)
                << parameter(":AAThreshold", p->aaThreshold)
                << parameter(":ThresholdMethod", (int)p->aaThresholdMethod)
                << parameter(":StartDate", p->origStartDate.toJulianDay())
                << parameter(":PortfolioID", p->id)
        );
    }
}
